var/spack/repos/builtin/packages/elpa/package.py

```diff
@@ -76,7 +75,14 @@ def setup_environment(self, spack_env, run_env):
         spack_env.set('SCALAPACK_LDFLAGS', spec['scalapack'].libs.joined())
 
     def configure_args(self):
-        options = []
+        # TODO: set optimum flags for platform+compiler combo, see
+        # https://github.com/hfp/xconfigure/tree/master/elpa
+        # also see:
+        # https://src.fedoraproject.org/cgit/rpms/elpa.git/
+        options = [
+            'FCFLAGS=-O3 -march=native -ffree-line-length-none',
+            'CFLAGS=-O3 -march=native'
```

Comment:
I am ccing @lee218llnl as I remember we had a discussion (was it `cp2k`?) where flags like `-march=<something>` were considered controversial. Also isn't `-march` GCC specific?

Reply:
for the record, without these flags configure fails testing `sse` related assembly. I can wrap it into `if spec.satisfies('%gcc')`.

Reply:
> Also isn't -march GCC specific?

i don't think so, it certainly works with `clang` and `intel`.

Reply:
> i don't think so, it certainly works with clang and intel.

Thanks, didn't have `intel` at hand to double check.

Reply:
> didn't have intel at hand to double check.

i also don't :smile: but I know for sure that my colleagues use this flag with `intel`.

Reply:
The Intel compiler does accept the -march option. The controversy with -march=native is for sites with a shared file system that is mounted on different architectures. For example, we have a system that supports AVX2, so compiling with the Intel compiler and -march=native enables AVX2 instructions. If I then run on older hardware that doesn't have AVX2, I see this:

Please verify that both the operating system and the processor support Intel(R) MOVBE, F16C, FMA, BMI, LZCNT and AVX2 instructions.

Reply:
@lee218llnl anything in spack like `if sepc.satisfies(platform ==)` i should add here in order to avoid the issue you described?

Reply:
I don't think the platform or anything else currently in the arch specification will help here. You could try using the `-mtune=generic` flag instead of `-march=native`, but you wouldn't get optimal performance. @becker33 do you have any suggestions for how to deal with vectorization flags?

Reply:
There isn't anything currently in the arch specification. I am working on this in #3206, but there's still some work to do there. That PR will make the target be as specific as the gcc arguments to `-march`.

Reply:
I think I will make a variant which is true by default. This shall work for majority of cases